### PR TITLE
Update documentation for FatalError handlers

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Compilers/Core/Portable/InternalUtilities/FatalError.cs [hunk 1]
```diff
@@ -130,11 +134,18 @@ public static bool ReportAndCatchUnlessCanceled(Exception exception)
         }
 
         /// <summary>
-        /// Use in an exception filter to report a non-fatal error. 
-        /// Calls <see cref="NonFatalHandler"/> unless the operation has been cancelled. 
-        /// The exception isn't passed through (the method returns true).
+        /// Use in an exception filter to report a non-fatal error (by calling <see cref="NonFatalHandler"/>) and catch
+        /// the exception, unless the operation was cancelled at the request of a caller.
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/InternalUtilities/FatalError.cs:138]: >at the request of a caller. [](start = 62, length = 27)

I think the summary should clearly explain how the method detects that.

Comment (sharwell) [reply src/Compilers/Core/Portable/InternalUtilities/FatalError.cs:138]: Quick Info shows both `<summary>` and `<remarks>`. I was hoping this part was clear:
image

Comment (AlekseyTs) [reply src/Compilers/Core/Portable/InternalUtilities/FatalError.cs:138]: I personally don't see any benefit in splitting this info between different sections of the doc comment. Is there a reason?

---
In reply to: [604244461](https://github.com/dotnet/roslyn/pull/52253#discussion_r604244461) [](ancestors = 604244461)

Comment (AlekseyTs) [reply src/Compilers/Core/Portable/InternalUtilities/FatalError.cs:138]: In my opinion the section that introduces a term "cancelled at the request of a caller" should provide an explanation of the term.

---
In reply to: [604287824](https://github.com/dotnet/roslyn/pull/52253#discussion_r604287824) [](ancestors = 604287824,604244461)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 4)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off.

**Review by cston — APPROVED** [review]

**Pull request merged**